assets/js/hooks/useScreenplayContext.tsx

```diff
@@ -0,0 +1,226 @@
+import * as React from "react";
+import { createGenericContext } from "../utils/createGenericContext";
+import { Place } from "../models/place";
+import { Alert } from "../models/alert";
+import { ScreensByAlert } from "../models/screensByAlert";
+import { useReducer } from "react";
+import {
+  MODES_AND_LINES,
+  SCREEN_TYPES,
+  STATUSES,
+} from "../constants/constants";
+import { BannerAlert } from "../components/Dashboard/AlertBanner";
+
+interface Props {
+  children: React.ReactNode;
+}
+
+type DirectionID = 0 | 1;
+
+interface FilterValue {
+  label: string;
+  ids: string[];
+  color?: string;
+}
+
+interface PageProps {
+  sortDirection: DirectionID;
+  modeLineFilterValue: FilterValue;
+  screenTypeFilterValue: FilterValue;
+  statusFilterValue: FilterValue;
+}
+
+interface StateContextType {
+  places: Place[];
+  alerts: Alert[];
+  screensByAlertMap: ScreensByAlert;
+  placesPageDirectionId: DirectionID;
+  placesPage: PageProps & {
+    showScreenlessPlaces: boolean;
+    activeEventKeys: string[];
+  };
+  alertsPage: PageProps;
+  bannerAlert?: BannerAlert;
+}
+
+const reducer = (state: any, action: any) => {
```

Comment:
It seems like we always send a string for action. Should we type check it as a string?

Reply:
I added a type for the reducer action so all potential dispatch types are documented.